### PR TITLE
144 parsing json files for ingredient input through barcode

### PR DESCRIPTION
**Pull request by JeannedeM**
Goals :

- [x]     Parse Json string properly.
- [x]     Modify what informations are displayed to user after barcode scanning.
- [x]     Make those informations more user friendly based on the json parsing.
- [x]     Scanned product are added to the ingredient list.

### CODE REVIEW

**Comment by JeannedeM** [comment]
Once again we decided to not have full coverage on this function. We cannot add barcode images in the CI Emulator hence it's almost impossible to test the function.....

File: app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt
Comment (mche3kek) [inline app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt:1]: An ingredient will always have a default metadata ? Why so ?

File: app/src/main/java/com/android/feedme/ml/Barcode.kt
[the remark this answers is not in the document]
Comment (JeannedeM) [reply app/src/main/java/com/android/feedme/ml/Barcode.kt:1]: Good question ! it's an information that i receive when i make a request to the Api, basically it tells me if the API has information about a product. Sometimes the API also has partial informations only.

File: app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt
Comment (JeannedeM) [reply app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt:1]: It's only temporary until Sylvain implements the functionnality to add unknown Ingredient to the data base in his next PR.

File: app/src/main/java/com/android/feedme/ml/Barcode.kt
[the remark this answers is not in the document]
Comment (mche3kek) [reply app/src/main/java/com/android/feedme/ml/Barcode.kt:1]: Okay, so it is useful for deciding if we output an error or not, thanks for the clarificattion !

File: app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt
Comment (mche3kek) [reply app/src/main/java/com/android/feedme/model/viewmodel/CameraViewModel.kt:1]: Oh I see, thanks for the precision

**Review by mche3kek — APPROVED** [review]
Everything looks good to me :)

**Pull request merged**